facilitator/src/idl.rs

```diff
@@ -70,13 +70,25 @@ pub trait Header: Sized {
     fn write<W: Write>(&self, writer: &mut W) -> Result<(), Error>;
 }
 
-pub trait Packet: Sized {
+pub trait Packet: Sized + TryFrom<Value, Error = Error> {
     /// Reads and parses a single Packet from the provided avro_rs::Reader. Note
     /// that unlike other structures, this does not take a primitive
     /// std::io::Read, because we do not want to create a new Avro schema and
     /// reader for each packet. The Reader must have been created with the
     /// schema returned from Packet::schema.
-    fn read<R: Read>(reader: &mut Reader<R>) -> Result<Self, Error>;
+    fn read<R: Read>(reader: &mut Reader<R>) -> Result<Self, Error> {
```

Comment:
I think this method is only used in this module's tests. Everything else is using `Packet::try_from`. Can we delete this method from the trait and fix the tests to use `try_from`?

Reply:
Oh, good catch; removed `Packet::read` from the trait & updated tests.